# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieServiceParent.cpp
# Commit: b27d9c0f5596
# Full Hash: b27d9c0f55969603a43bef741a933eff136f28ad
# Author: edgul <edgul@mozilla.com>
# Date: 2022-08-30 21:04:05
# Description:
#   Bug 1787122 - Fixed assert crash when attempting to get base domain from a malformed URL. r=kershaw
#   
#   Testing will be completed in: https://bugzilla.mozilla.org/show_bug.cgi?id=1788080
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155878
# ==============================================================================

diff -r ff5ca5a41f44 -r b27d9c0f5596 netwerk/cookie/CookieServiceParent.cpp
--- a/netwerk/cookie/CookieServiceParent.cpp	Tue Aug 30 16:17:31 2022 +0000
+++ b/netwerk/cookie/CookieServiceParent.cpp	Tue Aug 30 16:49:43 2022 +0000
@@ -82,6 +82,7 @@
 
 bool CookieServiceParent::CookieMatchesContentList(const Cookie& cookie) {
   nsCString baseDomain;
+  // CookieStorage notifications triggering this won't fail to get base domain
   MOZ_ALWAYS_SUCCEEDS(CookieCommons::GetBaseDomainFromHost(
       mTLDService, cookie.Host(), baseDomain));
 
@@ -135,8 +136,12 @@
     nsIURI* uri, const OriginAttributes& originAttrs) {
   nsCString baseDomain;
   bool requireAHostMatch = false;
-  MOZ_ALWAYS_SUCCEEDS(CookieCommons::GetBaseDomain(mTLDService, uri, baseDomain,
-                                                   requireAHostMatch));
+
+  // prevent malformed urls from being added to the cookie list
+  if (NS_WARN_IF(NS_FAILED(CookieCommons::GetBaseDomain(
+          mTLDService, uri, baseDomain, requireAHostMatch)))) {
+    return;
+  }
 
   CookieKey cookieKey(baseDomain, originAttrs);
   bool& allowSecure = mCookieKeysInContent.LookupOrInsert(cookieKey, false);
